scanpy/tests/test_pca.py

```diff
@@ -318,6 +318,20 @@ def test_mask_highly_var_error(array_type):
         sc.pp.pca(adata, use_highly_variable=True)
 
 
+def test_pca_backed(backed_adata):
+    with pytest.raises(
+        NotImplementedError,
+        match=f"PCA is not implemented for matrices of type {type(backed_adata.X)} from layers with chunked as False",
+    ):
+        sc.pp.pca(backed_adata)
```

Comment:
can you unify all the tests that follow this exact template?

```py
@pytest.mark.parametrize(
    ("name", "func", "msg"),
    [
        pytest.param("PCA", sc.pp.pca, " from layers with chunked as False", id="pca"),
        pytest.param("regress_out", partial(sc.pp.regress_out, keys=["n_counts", "percent_mito"]), "", id="regress_out"),
        ...
    ]
)
def test_backed_error(backed_adata, name, func, msg):
    regex = re.escape(f"{name} is not implemented for matrices of type {type(backed_adata.X)}{msg}")
    with pytest.raises(NotImplementedError, match=regex):
        func(backed_adata)

Reply:
Where to put it though?  New file?

Reply:
that's what i did

Reply:
Yup, that’s what I meant!